### ISSUE TITLE
International keyboard does not always activate

### ISSUE THREAD

**Issue by Heretushi** [koush/vysor.io#555]
**Describe the bug**
International keyboard don't activate on Vysor startup about 1/3 of the time

**Attach a Vysor Bug Report Log**
1. First, trigger the bug.
2. **Leave the window with the bug open.** Do not close it, or the bug data will be lost.
3. To generate a bug report, open the main device window if it is not open already. Click the Bug Report button along the bottom. Please note that this will contain your email address. This is necessary for diagnosing licensing/purchase issues.
4. **ISSUES WITHOUT BUG REPORTS MAY BE IMMEDIATELY CLOSED.** The bug report contains internal logging from the app.
https://gist.github.com/VysorLogs/ae839157904c97ec56415b72782f4117

**To Reproduce**
1. Start Vysor.
2. Make sure International Keyboard option is on.
3. Connect the phone, click View.
4. Try to type any accented character in Vysor. They don't work.
5. Close the phone window, click "Reload Vysor" in Vysor window, try again and International Keyboard will probably be active.

**Expected behavior**
I expect International Keyboard to be active at every try.

**Screenshots**
N/A

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Vysor Version: 3.0.95

**Android (please complete the following information):**
 - Device: Pixel 3
 - OS: Android 10
 - Vysor Bug Report Link: https://gist.github.com/VysorLogs/ae839157904c97ec56415b72782f4117

**Android Logcat**
Need instruction to produce if needed.

**Additional context**
N/A

**Comment by koush** [comment]
How are you typing accented characters? I can, for example, change my keyboard type in Mac to Hebrew, and the different character set (as well as left to right) works.
Let me know what exactly you're doing on the Windows side, and I'll take a look. I used key character map in the past to test it. Is that what you're using?

When you say it doesn't work, do you mean not at all? Or occasionally doesn't start the Vysor Keyboard? Also, what app are you typing into?

To summarize:
 * How are you typing these characters in Windows
 * What app are you typing into in Android (I use the search bar or Android Chrome to test)

**Comment by Heretushi** [comment]
Hi, 
So my keyboard is mapped with the "French (Canadian)" map in Windows. I'm not typing any differently than any other time. When it does not work, in every Android app, I can type any character I can normally type when NOT using the international keyboard option. Then, when I close, reload and reopen Vysor without touching any other setting anywhere, the keyboard does produce accented characters.

For testing purpose, I usually type in the Google search box OR in Messenger. Again, I'm not touching anything on the Android side or the Windows side. I just stop typing, open Vysor main window, hit "Reload Vysor", then click "View", then start typing in the app right where I was and it (usually) works. :) Is it clear enough ? If you want, I can try producing a video of the bug if I'm persistent enough.

**Comment by koush** [comment]
No need at the moment, I think I have enough info to reproduce this hopefully.

**Comment by Heretushi** [comment]
FYI, I did encounter the error this morning and took a 45 seconds video. I could share the URL privately koush since it shows my work environment. Thanks.

**Comment by Heretushi** [comment]
New information : I got the problem once more this morning. Instead of "reloading" Vysor altogether, I just tried to close the Vysor "device windows" and reopen it with the view button. It worked. It then seems to be that sometime, the keyboard on Android does not trigger/ is not enable correctly.

**Comment by koush** [comment]
Hi @Heretushi 
I want you to know that I'm not ignoring you.

I live in Seattle. We're rioting, I've been busy protesting :)

**Comment by Heretushi** [comment]
@koush , please, fight for what is right and don't worry about my little quibbles. I will close the window and reopen it gladly if it leads to a strong message being delivered on matter so much more important.

My only concern : stay safe.

**Comment by Dimergon** [comment]
I have more or less the same problem as OP, except my keyboard refuses to work at all. Doesn't matter if I've enabled international keyboard or not, Vysor simply refuses to let me input text from my keyboard to my Huawei P30 Pro, not wirelessly nor with a cable. I've tried rebooting both PC and phone, I've tried different versions of Vysor, I've reloaded Vysor to try again, but without any luck so far. 

The Vysor keyboard on my phone is simply dead, the only keys that work are the keybindings from the config (arrow keys, enter, delete, home, escape etc.).

Any clue as to why I cannot input text anywhere with the Vysor input? I can't even enter the settings menu for the keyboard, when I press it I just get instantly kicked out to the previous menu screen.

>Bugreport: https://gist.github.com/61fc6555a955b17e12d766b06aeb6118

Hope I can get some help with this, because otherwise those $10.00 I spent on the annual subscription was a total waste

**Comment by Heretushi** [comment]
This part is particularly interesting to me : 

> I can't even enter the settings menu for the keyboard, when I press it I just get instantly kicked out to the previous menu screen.

So your saying that on your phone, even without it being plugged into your PC, you can't get into the Vysor Keyboard app on your phone?

If your phone doesn't even want you to open the Vysor Keyboard Android App, I feel it's pretty normal the Vysor Desktop App won't "fully work". 

Can you clarify ? :)

**Comment by Dimergon** [comment]
Sure. Have a look at [this](https://streamable.com/4aoaoc) and [this](https://streamable.com/qunbdd). The part where I get kicked out of the Vysor Keyboard settings happens regardless of physical or wireless connection.

Hoping for some insightful answers here, It really sucks not being able to use accented characters with this app.

**Comment by Dimergon** [comment]
Anyone?

**Comment by Heretushi** [comment]
Have you tried having your phone in english for testing purposes ?

**Comment by Dimergon** [comment]
Yes I have, but to no avail. The "Vysor"-keyboard input does not react at all when I try to type, but keys like escape, enter, backspace and arrow keys do work. The Google Swiftkey keyboard does work when I try to type, but does not react to me pressing accented keys like "Æ" "Ø" "Å" (which are essential to the Norwegian language), So I'm completely lost here. Doesn't matter if my phones language is in english or norwegian, doesn't matter if "International keyboard" is enabled or disabled, doesn't matter if I use Swiftkey or Vysor input methods, I cannot get it to work as intended. The Vysor input method does not work at all, but the Swiftkey one works as long as I'm using the English language.

**Comment by Heretushi** [comment]
I understand that Vysor is worse than SwiftKey in your situation. But the fact that SwiftKey also do not work properly is telling me something is wrong with your phone. SwiftKey must have some kind of fallback in place that Vysor does not have, either because of ressources to implement (time, money, knowledge, knowhow, etc) or because of technical restrictions. I know this answer does no good to you, but if I were in your place, I would spend energy trying to get an actual big name Android keyboard to work properly before trying to get support from Vysor. Sadly, these are shaky times in the US and Vysor is still a one man show. Hope you find a solution! 👍

**Comment by koush** [comment]
@Heretushi Can you shoot me a video of this issue? I've not been able to reproduce it.

**Comment by Heretushi** [comment]
Sure, will do!

**Comment by Heretushi** [comment]
Tried to repro on my work setup. Since it's intermittent, I could not. Will try all day long, hope I will get it.

**Comment by Heretushi** [comment]
I seriously tried to repro, could not. :| But really, it's not complicated : it's everything as normal, but when typing accented letters, they do not appear accented. 
Since it's been weeks and I cannot repro, you are free to close the issue koush.
Thanks.

**Comment by koush** [comment]
I'll try leaving the international running for a while and see if it repros after some time.

**Comment by koush** [comment]
Seems related. Found a bug where international keyboard was working by pure luck on multicharacter languages. https://github.com/koush/vysor.io/issues/684

**Comment by koush** [comment]
Using https://github.com/koush/vysor.io/issues/684 as a rollup bug for all keyboard issues.

**Comment by gotmachine** [comment]
Having this issue too.

Desktop (please complete the following information):

    OS: Windows 10
    Vysor Version: 4.0.29

Android (please complete the following information):

    Device: Motorola Moto-E 4G 2nd gen
    OS: Android 6.0
    Vysor Bug Report Link: https://gist.github.com/VysorLogs/29a6d963bc54a5c7b820c9f051f1724b
    
Using a french keyboard with the phone connected trough USB, I've managed to get the Vysor keyboard to accept accentuated characters like é or à once, other characters are accepted fine (with or without the international option checked). I'm not sure exactly how I managed to get it working, but I remember switching the international keyboard option multiple times, re-plugging the phone, relaunching Vysor, switching between the Gboard and Vysor keyboards on the phone. Now I can't manage to get it working again.

**Comment by clubtech** [comment]
I started to have this issue as well with the international keyboard where it will not process ANY input (even English) intermittently.
About 2/3 of the time when connecting there is no issue at all (using both English and Hebrew letters) but about 1/3 of the time the keyboard will show up in the Android's top bar but it won't process any input at all, no matter the language.

The more interesting thing is that I am using Vysor with two Windows 10 computers, both running the exact same version of Windows (latest version 19042.804). One Laptop is from HP and the other from Dell. The one from HP never has this issue. The one from Dell does this about 1/3 of the time.

When this happens on the Dell, reconnecting to Vysor does NOT solve the issue. It's like something is getting stuck on the phone side. Only restarting the phone or waiting a few hours solves this.

I am using both computers with the Galaxy S21 Ultra.

It is extremely weird that one laptop will never have this issue while the other has this about 1/3 of the time.
I have a MacBook as well so I will test this using a Mac and see if it happens or not.

I'll post a bug log the next time this happens.

**Comment by clubtech** [comment]
Here is a bug report for the issue in mu comment above.
Connected from the Dell laptop to the phone, the international keyboard launched but is not accepting input in any language: 

https://gist.github.com/VysorLogs/324706f082ca3baf6906c53925ff4e5b

**Comment by koush** [comment]
> The more interesting thing is that I am using Vysor with two Windows 10 computers, both running the exact same version of Windows (latest version 19042.804). One Laptop is from HP and the other from Dell. The one from HP never has this issue. The one from Dell does this about 1/3 of the time.


Is this the same phone exhibiting different behavior on different computers?

**Comment by clubtech** [comment]
Yes, the exact same phone with two different laptops.
This never happens on the HP. It does happen 1/3 of the time on the Dell.
Both laptops are on the same wifi network.
Both were clean installed.
Both have the same language settings (English US is the main, Hebrew is second).
It just happened again with the Dell. Here's another log: https://gist.github.com/VysorLogs/f94d504b7b106c80e7136f7fd373966b

**Comment by koush** [comment]
@clubtech I noticed you're on a wireless connection. can you confirm this is also occurring on a wired connection?

**Comment by clubtech** [comment]
@koush yes, I just connected the phone via USB and the same keyboard issue just happened.
Log: https://gist.github.com/VysorLogs/c055ce230a2b7f952189596a048b55b6

The thing is sometimes when this happens, if I just wait 10-20 minutes without doing anything, sometimes (but not always) the problem will correct itself the keyboard will be responding to input again. Again, so far this has only happened to me on the Dell machine.

**Comment by Guns1970** [comment]
Hi everyone,

had a similar issue (HP desktop, W10 20H2 V2 / Vysor 4.0.29 - Oneplus 6 Android 10 / App v 4.0.0) : Int'l keyboard frequently just not responding at all ... mostly after long idle state, sometimes right on startup.

I found a workaround that works 100% for me -> clear the app cache on my smartphone (I don't even need to reload app on my PC). If that can help ...

PS also using the app flawlessly on my Acer Swift laptop.

**Comment by clubtech** [comment]
@Guns1970 sadly clearing the cache does not work for me.
The issue still persists with one of my laptops. Super weird.

**Comment by dieskim** [comment]
These issues are still around for me also. Same as before - see #684 and #555

Currently the best way for me seems to be to FORCE STOP the app on the phone - then wait a min and then set the phone back to the Vysor Keyboard

What can we provide you with @koush to help us get this fixed?

**Comment by dieskim** [comment]
@koush - I think all of these are related 
#555 #664 #736 #715

**Comment by koush** [comment]
@dieskim Hi,

I've been running international keyboard myself for several weeks with no issues. Other bug reports have indicated that this occurs only on some machines and not others, with the exact same phone. I'm completely at a loss as to what the issue could be. I'm sorry for everyone's frustration.

I'm adding some code to retrieve the logging from the Android side (the logcat), to see precisely what is happening with the IME keyboard service. Currently the Android side logs are not collected. I'm hoping with this additional logging, I can track down why the keyboard service and/or connection is dying.

**Comment by dieskim** [comment]
@koush - ok good to know you can now get those extra logs. This is happening for me on different devices for sure. How can we help you from our side now to get these logs? Shall we capture the logs when it happens and send to you?

**Comment by koush** [comment]
Yes, if you are familiar with how to capture android logcats, after the error occurs, capture it, and attach it here as a gist link.

**Comment by dieskim** [comment]
@koush - sure but I thought you said it will capture those automatically now?

**Comment by koush** [comment]
I will, it's something I will try to implement and roll out this week

**Comment by koush** [comment]
Well, I implemented the logging and it turned out to not be that useful. There's a size limit on the logs I can upload, and the logs tend to go over it, so I need to trim them down. Android logcat is extremely noisy.

I have a a possible wild guess fix, and other logging mechanisms to track why the IME would stop responding. Will be in the next update.

**Comment by dieskim** [comment]
ok I will keep an eye on it after the new update - thanks

**Comment by koush** [comment]
@dieskim Hi, I've updated the app last week to provide more logs. A new log of the error would be greatly helpful in figuring out what is getting stuck/killed and causing this problem.

**Comment by koush** [comment]
I appreciate the continued help btw.

**Comment by koush** [comment]
Thanks to some live debugging from @dieskim I believe this is fixed. It was cause by the message queue of the keyboard to be backed up and delayed. Please confirm fix.

I believe this is fixed in 4.0.36.
